server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/fileformats/meta/StarTreeMetadataWriter.java

```diff
@@ -171,4 +173,24 @@ private static void writeMeta(
         metaOut.writeVLong(dataFileLength);
 
     }
+
+    private static byte docValuesByte(DocValuesType type) {
+        switch (type) {
+            case NONE:
+                return 0;
+            case NUMERIC:
+                return 1;
+            case BINARY:
+                return 2;
+            case SORTED:
+                return 3;
+            case SORTED_SET:
+                return 4;
+            case SORTED_NUMERIC:
+                return 5;
+            default:
+                // BUG
+                throw new AssertionError("unhandled DocValuesType: " + type);
```

Comment:
IllegalStateException? Or is there a precedent to use AssertionError

Reply:
This is actually taken straight from Lucene